src/xdp/trn_transit_xdp_hardware_offload.c

```diff
@@ -0,0 +1,580 @@
+	ep = bpf_map_lookup_elem(&endpoints_map, &epkey);
+
+	/* Don't respond to arp if endpoint is not found, or it is local to host */
+	if (!ep || ep->hosted_iface != -1 ||
+	    pkt->inner_arp->ar_op != bpf_htons(ARPOP_REQUEST)) {
+
+		return trn_switch_handle_pkt(pkt, *sip, *tip, *sip);
+	}
+
+	/* Respond to ARP */
+	pkt->inner_arp->ar_op = bpf_htons(ARPOP_REPLY);
+	trn_set_arp_ha(tha, sha);
+	trn_set_arp_ha(sha, ep->mac);
+
+	__u32 tmp_ip = *sip;
+	*sip = *tip;
+	*tip = tmp_ip;
+
+	/* Set the sender mac address to the ep mac address */
+	trn_set_src_mac(pkt->inner_eth, ep->mac);
+
+	if (ep->eptype == TRAN_SIMPLE_EP) {
+		/*Get the remote_ep address based on the value of the outer dest IP */
+		epkey.tunip[0] = 0;
+		epkey.tunip[1] = 0;
+		epkey.tunip[2] = ep->remote_ips[0];
+		remote_ep = bpf_map_lookup_elem(&endpoints_map, &epkey);
+
+		if (!remote_ep) {
+
+			return XDP_DROP;
+		}
+
+		/* For a simple endpoint, Write the RTS option on behalf of the target endpoint */
+		pkt->rts_opt->rts_data.host.ip = ep->remote_ips[0];
+		__builtin_memcpy(pkt->rts_opt->rts_data.host.mac,
+				 remote_ep->mac, 6 * sizeof(unsigned char));
+	} else {
+
+		trn_reset_rts_opt(pkt);
+	}
+
+	/* We need to lookup the endpoint again, since tip has changed */
+	epkey.tunip[2] = *tip;
+	ep = bpf_map_lookup_elem(&endpoints_map, &epkey);
+
+	return trn_switch_handle_pkt(pkt, *sip, *tip, *sip);
+}
+
+static __inline int trn_process_inner_eth(struct transit_packet *pkt)
+{
+	pkt->inner_eth = (void *)pkt->geneve + pkt->gnv_hdr_len;
+	pkt->inner_eth_off = sizeof(*pkt->inner_eth);
+
+	if (pkt->inner_eth + 1 > pkt->data_end) {
+
+		return XDP_ABORTED;
+	}
+
+	/* ARP */
+	if (pkt->inner_eth->h_proto == bpf_htons(ETH_P_ARP)) {
+
+		return trn_process_inner_arp(pkt);
+	}
+
+	if (pkt->eth->h_proto != bpf_htons(ETH_P_IP)) {
+
+		return XDP_DROP;
+	}
+
+	return trn_process_inner_ip(pkt);
+}
+
+static __inline int trn_process_geneve(struct transit_packet *pkt)
+{
+	pkt->geneve = (void *)pkt->udp + sizeof(*pkt->udp);
+	if (pkt->geneve + 1 > pkt->data_end) {
+
+		return XDP_ABORTED;
+	}
+
+	if (pkt->geneve->proto_type != bpf_htons(ETH_P_TEB)) {
+
+		return XDP_PASS;
+	}
+
+	pkt->gnv_opt_len = pkt->geneve->opt_len * 4;
+	pkt->gnv_hdr_len = sizeof(*pkt->geneve) + pkt->gnv_opt_len;
+	pkt->rts_opt = (void *)&pkt->geneve->options[0];
+
+	if (pkt->rts_opt + 1 > pkt->data_end) {
+
+		return XDP_ABORTED;
+	}
+
+	if (pkt->rts_opt->opt_class != TRN_GNV_OPT_CLASS) {
+
+		return XDP_ABORTED;
+	}
+
+	// TODO: process options
+	pkt->scaled_ep_opt = (void *)pkt->rts_opt + sizeof(*pkt->rts_opt);
+
+	if (pkt->scaled_ep_opt + 1 > pkt->data_end) {
+
+		return XDP_ABORTED;
+	}
+
+	if (pkt->scaled_ep_opt->opt_class != TRN_GNV_OPT_CLASS) {
+
+		return XDP_ABORTED;
+	}
+
+	return trn_process_inner_eth(pkt);
+}
+
+static __inline int trn_process_udp(struct transit_packet *pkt)
+{
+	/* Get the UDP header */
+	pkt->udp = (void *)pkt->ip + sizeof(*pkt->ip);
+
+	if (pkt->udp + 1 > pkt->data_end) {
+
```

Comment:
remove blank line. please follow the conventions used in the rest of the xdp code